Document caveat re large Typst projects (#4907)
typst/typst: docs/guides/guide-for-latex-users.md

  break will be created if this spot would be a natural page break anyways. You
  can also use `[#v(1fr)]` to distribute space on your page. It works quite
  similar to LaTeX's `\vfill`.

- **Separation of `import` and `include` behaviours.** `\include`ing a file in LaTeX
  has two behaviours that have been separated in Typst: imports LaTeX macros,
  and adds the document content to the parent file. In Typst: `#import`ing will
  add definitions (equivalent to LaTeX macros) to the parent file,
  but not add document content; `#include`ing will add document content
  to the parent file, but will not import definitions.
  The syntax for including document content is simply `#include "file.typ"`,
  but the syntax for importing definitions provides more flexibility:
  `#import "file.typ": *` will import all of `file.typ`'s definitions;
  `#import "file.typ": x, y, template` will import only the named definitions.
  It is of particular note for those familiar with LaTeX that
  in LaTeX, a file only needs to be `\include`d once;
  in Typst, your global definitions
  and global setups need to be placed in a separate file,
  and that file is then `#import`ed and/or `#include`d

>   into all of your project's nested files in addition to the main file.

Reviewer: ```suggestion
  into the files that make use of these definitions.
```

Just like with every good programming language, you only need to import stuff to where you use it.

Author: This would call for three boilerplate lines of code then.
1. Any `#include`s.
2. `#import` of definitions specific to the file.
3. `#include`/`#import` of setups that are used in every file, e.g. default font.

  break will be created if this spot would be a natural page break anyways. You
  can also use `[#v(1fr)]` to distribute space on your page. It works quite
  similar to LaTeX's `\vfill`.

> - **Boilerplate for large scale projects.** LaTeX definitions are available in
>   included files. This is not the case in Typst, where you'll need your definitions
>   in a file of its own, and manually `import` them into each of your project's
>   `include`d files. Alternatively your Typst project can be in one very large file.

Reviewer 2: This doesn't seem like the best wording for it, and an example would be nice as well. I'm not sure this belongs in "Limitations" either, since the feature does exist in Typst. Here's an initial idea on how to improve it, but larger-scale changes (e.g. moving to another section or merging with the existing text on this) may be needed.

```suggestion
- **Separation between `import` and `include`.** LaTeX macro definitions in files are
  automatically available in documents which `\include` those files' contents. In Typst,
  including a file's contents (done through `#include "file.typ"`) is separate from
  importing its definitions, which must be done with a separate
  `#import "file.typ": definitions` command. With it, you can either list the definitions
  to import individually, which is more usual when they are few (e.g.
  `#import "file.typ": x, y, template`), or, to fully mimic LaTeX's behavior, you may use a
  wildcard import to import all definitions available in another file
  (e.g. `#import "file.typ": *`).
```

Author: Thank you so much. I did want to get the ball rolling on this caveat because this limitation of Typst frustrates me, and I wanted it to be in the list of limitations for LaTeX users, particularly since there are at least three different issues reported by users on this, so the documentation is definitely not clear enough.

Reviewer 2: Understood, thank you for your input. Would be nice to see what other LaTeX users think of this change as well.

Reviewer 3: As a long-time LaTeX user, I find this very helpful. I stumbled on this a few times and, while it's not (at least from my experience) a major blocker for moving from LaTeX to Typst, making it as explicit as possible can probably make the experience even smoother than it already is.
